src/linter/root.go

```diff
@@ -1214,15 +1214,15 @@ func (d *RootWalker) enterFunction(fun *stmt.Function) bool {
 	if sideEffectFreeFunc(d.scope(), d.st, nil, fun.Stmts) {
 		funcFlags |= meta.FuncPure
 	}
-	d.meta.Functions[nm] = meta.FuncInfo{
+	meta.AddFunctionToMap(d.meta.Functions, nm, meta.FuncInfo{
```

Comment:
You're not holding the metadata lock here.

Reply:
Sorry, got confused. You're right, you're modifying the local map here. Maybe rename it to something like 'AppendFunction' and return the modified map so that it's clearer that you're not modifying the global state in meta?